source/fuzz/transformation_vector_dynamic.cpp

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2020 André Perez Maselco
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "source/fuzz/transformation_vector_dynamic.h"
+
+#include "source/fuzz/fuzzer_util.h"
+#include "source/fuzz/instruction_descriptor.h"
+
+namespace spvtools {
+namespace fuzz {
+
+TransformationVectorDynamic::TransformationVectorDynamic(
+    const spvtools::fuzz::protobufs::TransformationVectorDynamic& message)
+    : message_(message) {}
+
+TransformationVectorDynamic::TransformationVectorDynamic(
+    uint32_t instruction_result_id) {
+  message_.set_instruction_result_id(instruction_result_id);
+}
+
+bool TransformationVectorDynamic::IsApplicable(
+    opt::IRContext* ir_context,
+    const TransformationContext& transformation_context) const {
+  // |instruction| must be defined and must be an OpCompositeExtract or an
+  // OpCompositeInsert instruction.
+  auto instruction =
+      ir_context->get_def_use_mgr()->GetDef(message_.instruction_result_id());
+  if (!instruction || (instruction->opcode() != SpvOpCompositeExtract &&
+                       instruction->opcode() != SpvOpCompositeInsert)) {
+    return false;
+  }
+
+  // The composite must be a vector.
+  auto composite_instruction =
+      ir_context->get_def_use_mgr()->GetDef(instruction->GetSingleWordInOperand(
+          instruction->opcode() == SpvOpCompositeExtract ? 0 : 1));
+  if (!ir_context->get_type_mgr()
+           ->GetType(composite_instruction->type_id())
+           ->AsVector()) {
+    return false;
+  }
```

Comment:
I suggest you introduce a public static helper method in TransformationVectorDynamic that checks whether an instruction is suitable for being made dynamic - then you can avoid the duplication between here and the fuzzer pass.

Reply:
Done.